bucket.go

```diff
@@ -309,15 +311,83 @@ func (b *Bucket) DeleteBucket(key []byte) (err error) {
 	}
 
 	// Remove cached copy.
-	delete(b.buckets, string(key))
+	delete(b.buckets, string(newKey))
 
 	// Release all bucket pages to freelist.
 	child.nodes = nil
 	child.rootNode = nil
 	child.free()
 
 	// Delete the node if we have a matching key.
-	c.node().del(key)
+	c.node().del(newKey)
+
+	return nil
+}
+
+// MoveBucket moves a sub-bucket from the source bucket to the destination bucket.
+// Returns an error if
+//  1. the sub-bucket cannot be found in the source bucket;
+//  2. or the key already exists in the destination bucket;
+//  3. or the key represents a non-bucket value;
+//  4. the source and destination buckets are the same.
+func (b *Bucket) MoveBucket(key []byte, dstBucket *Bucket) (err error) {
+	lg := b.tx.db.Logger()
+	lg.Debugf("Moving bucket %q", string(key))
+	defer func() {
+		if err != nil {
+			lg.Errorf("Moving bucket %q failed: %v", string(key), err)
+		} else {
+			lg.Debugf("Moving bucket %q successfully", string(key))
+		}
+	}()
+
+	if b.tx.db == nil || dstBucket.tx.db == nil {
```

Comment:
It seems like we should limit both src and dst are in that same db.

Reply:
yes, it's a good point. 

We can add protection something like below in `(*Bucket)MoveBucket`,

```
	if b.tx.db.Path() != dstBucket.tx.db.Path() {
		// return error
	}
```

We also need to add a test

@Elbehery can you take care of it in a following up PR?